mlos_bench/mlos_bench/optimizers/base_optimizer.py

```diff
@@ -46,7 +46,7 @@ def __init__(self, tunables: TunableGroups, service: Optional[Service], config:
         self._start_with_defaults: bool = bool(
             strtobool(str(self._config.pop('start_with_defaults', True))))
         self._max_iter = int(self._config.pop('max_iterations', 100))
-        self._opt_target = self._config.pop('optimization_target', 'score')
+        self._opt_target = str(self._config.pop('optimization_target', 'score'))
```

Comment:
Addresses an rnrelated mypy warning flagged from the return call type of `target` property.